task/discover_granules_s3.py

```diff
@@ -1,6 +1,7 @@
 import concurrent.futures
 import os
 import re
+from time import sleep
```

Comment:
This may be able to be removed. I didn't see anywhere it was being used within this file